code/controllers/subsystem/blackbox.dm

```diff
@@ -0,0 +1,274 @@
+	further calls to the same key will:
+	 	add or subtract from the saved value of the data key if it already exists
+		append the key and it's value if it doesn't exist
+	calls:	SSblackbox.record_feedback("tally", "example", 1, "sample data")
+			SSblackbox.record_feedback("tally", "example", 4, "sample data")
+			SSblackbox.record_feedback("tally", "example", 2, "other data")
+	json: {"data":{"sample data":5,"other data":2}}
+"nested tally"
+	used to track the number of occurances of structured semi-relational values i.e. the results of arcade machines
+	similar to running total, but related values are nested in a multi-dimensional array built
+	the final element in the data list is used as the tracking key, all prior elements are used for nesting
+	further calls to the same key will:
+	 	add or subtract from the saved value of the data key if it already exists in the same multi-dimensional position
+		append the key and it's value if it doesn't exist
+	calls: 	SSblackbox.record_feedback("nested tally", "example", 1, list("fruit", "orange", "apricot"))
+			SSblackbox.record_feedback("nested tally", "example", 2, list("fruit", "orange", "orange"))
+			SSblackbox.record_feedback("nested tally", "example", 3, list("fruit", "orange", "apricot"))
+			SSblackbox.record_feedback("nested tally", "example", 10, list("fruit", "red", "apple"))
+			SSblackbox.record_feedback("nested tally", "example", 1, list("vegetable", "orange", "carrot"))
+	json: {"data":{"fruit":{"orange":{"apricot":4,"orange":2},"red":{"apple":10}},"vegetable":{"orange":{"carrot":1}}}}
+	tracking values associated with a number can't merge with a nesting value, trying to do so will append the list
+	call:	SSblackbox.record_feedback("nested tally", "example", 3, list("fruit", "orange"))
+	json: {"data":{"fruit":{"orange":{"apricot":4,"orange":2},"red":{"apple":10},"orange":3},"vegetable":{"orange":{"carrot":1}}}}
+"associative"
+	used to record text that's associated with a value i.e. coordinates
+	further calls to the same key will append a new list to existing data
+	calls:	SSblackbox.record_feedback("associative", "example", 1, list("text" = "example", "path" = /obj/item, "number" = 4))
+			SSblackbox.record_feedback("associative", "example", 1, list("number" = 7, "text" = "example", "other text" = "sample"))
+	json: {"data":{"1":{"text":"example","path":"/obj/item","number":"4"},"2":{"number":"7","text":"example","other text":"sample"}}}
+
+Versioning
+	If the format of a feedback variable is ever changed, i.e. how many levels of nesting are used or a new type of data is added to it, add it to the versions list
+	When feedback is being saved if a key is in the versions list the value specified there will be used, otherwise all keys are assumed to be version = 1
+	versions is an associative list, remember to use the same string in it as defined on a feedback variable, example:
+	list/versions = list("round_end_stats" = 4,
+						"admin_toggle" = 2,
+						"gun_fired" = 2)
+*/
```

Comment:
Please quickly mention that the data is actually saved at the end of the round, not when the function is called.